### PR TITLE
Create your own Enumerable

### PR DESCRIPTION
**Pull request by oovillagran**
<h1>Hey there 👋</h1>

In this project, we learned how to use a module inside our class. For this, we created a class MyList and a module MyEnumerable. Our module MyEnumerable implemented a subset of the functionality of [Enumerable](https://ruby-doc.org/core-3.0.0/Enumerable.html).

**Project Requirements:**

- Check offical documentation about the following 3 methods in [Enumerable](https://ruby-doc.org/core-3.0.0/Enumerable.html). 
     - [description of #all? method](https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-all-3F)
     - [description of #any? method](https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-any-3F)
     - [description of #filter method](https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-filter)
- Create a class MyList that has an instance variable @list.
- In MyList implement a method #each that yields successive members of @list and uses the MyEnumerable module.
- Create a module MyEnumerable that implements the following methods (they should have the same funcionality as methods in [Enumerable](https://ruby-doc.org/core-3.0.0/Enumerable.html)):
     - #all?
     - #any?
     - #filter
- Each class and module should have a separate .rb file.

<h2>Thanks for your kind review and feedback 👏</h2>

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  include MyEnumerable
+
+  def initialize(*list)
+    @list = list
+  end
+
+  def each(&block)
+    @list.each(&block)
+  end
```
Comment (Ismailco) [inline my_list.rb]: - [x] Kindly make sure to create your own `each` method that yields all members of @list in succession. (You should not use the ruby `each` method but create your own)

File: my_enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    each do |item|
+      return true if yield(item)
+    end
+    false
+  end
+
+  def filter(&block)
+    select(&block)
+    # end
+  end
```
Comment (Ismailco) [inline my_enumerable.rb]: - [x] Kindly consider using the newly created `each` method here instead of the default `each` method

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @oovillagran, @naanmohammed, @Mylo16, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by oovillagran** [comment]
Thanks @Ismailco!

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @naanmohammed      👋

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀

image


## Highlights
- Good job with the use of GitHub features✔️ 
- Kudos on implementing the changes as requested ✅ 
## Optional Suggestions 💡
While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really enhance your application.
Cheers, and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread, and don't forget to tag me so I'll receive the notification.

As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by oovillagran** [comment]
Thanks @Shedrack-Sunday!

**Pull request merged**